assets/blocks/digital-wallets/content.js

```diff
@@ -6,48 +6,104 @@ import { useEffect, useState } from '@wordpress/element';
+import { getSquareServerData } from '../square-utils/utils';
+import {
+	useSquare,
+	usePaymentRequest,
+	useShippingContactChangeHandler,
+	useShippingOptionChangeHandler,
+	useGooglePay,
+	useApplePay,
+	usePaymentProcessing,
+} from './hooks';
 
-const Content = (props) => {
-	const [payments, setPayments] = useState(null);
-	const { onCheckoutFail } = props.eventRegistration;
-	const { onClose } = props;
+const Content = ({
+	billing,
+	shippingData,
+	onClick,
+	onClose,
+	onSubmit,
+	setExpressPaymentError,
+	emitResponse,
+	eventRegistration: { onPaymentSetup, onCheckoutFail },
+}) => {
+	const { needsShipping } = shippingData;
+	const payments = useSquare();
+	const paymentRequest = usePaymentRequest(payments, needsShipping);
+	const [googlePay, googlePayRef] = useGooglePay(payments, paymentRequest);
+	const [applePay, applePayRef] = useApplePay(payments, paymentRequest);
+	const [tokenResult, setTokenResult] = useState(false);
+	const [clickedButton, setClickedButton] = useState(null);
 
-	useEffect(() => {
-		const applicationId = getSquareServerData().applicationId;
-		const locationId = getSquareServerData().locationId;
+	useShippingContactChangeHandler(paymentRequest);
+	useShippingOptionChangeHandler(paymentRequest);
+	usePaymentProcessing(
+		payments,
+		billing,
+		clickedButton,
+		tokenResult,
+		emitResponse,
+		onPaymentSetup
+	);
 
-		if (!window.Square) {
+	useEffect(() => {
+		if (!clickedButton) {
 			return;
 		}
 
-		try {
-			const __payments = window.Square.payments(
-				applicationId,
-				locationId
-			);
-			setPayments(__payments);
-		} catch (e) {
-			console.error(e);
-		}
-	}, []);
+		setExpressPaymentError('');
+		onClick();
+
+		(async () => {
+			const __tokenResult = await tokenize(clickedButton);
+
+			if (!__tokenResult) {
+				onClose();
+				setClickedButton(null);
+			} else {
+				setTokenResult(__tokenResult);
+				onSubmit();
+			}
+		})();
+	}, [clickedButton]);
 
 	useEffect(() => {
-		const unsubscribeOnCheckoutFail = onCheckoutFail(() => {
+		const unsubscribe = onCheckoutFail(() => {
+			setClickedButton(null);
 			onClose();
 			return true;
 		});
+		return unsubscribe;
+	}, [onCheckoutFail]);
+
+	const isGooglePayDisabled =
+		getSquareServerData().hideButtonOptions.includes('google');
+	const isApplePayDisabled =
+		getSquareServerData().hideButtonOptions.includes('apple');
 
-		return unsubscribeOnCheckoutFail;
-	}, [onCheckoutFail]); // eslint-disable-line react-hooks/exhaustive-deps
+	const googlePayExpressButton = !isGooglePayDisabled && (
+		<div // eslint-disable-line jsx-a11y/click-events-have-key-events
+			tabIndex={0}
+			role="button"
+			ref={googlePayRef}
+			onClick={() => setClickedButton(googlePay)}
+		></div>
+	);
+
+	const applePayExpressButton = !isApplePayDisabled && (
+		<div // eslint-disable-line jsx-a11y/click-events-have-key-events
+			tabIndex={0}
+			role="button"
+			ref={applePayRef}
+			onClick={() => setClickedButton(applePay)}
+		></div>
+	);
```

Comment:
This ensures the Apple Pay button displays correctly and resolves a JS error.

You'll need to add something to ensure the button is hidden on render.


```suggestion
	const applePayExpressButton = !isApplePayDisabled && (
		<div // eslint-disable-line jsx-a11y/click-events-have-key-events
			tabIndex={0}
			role="button"
			ref={applePayRef}
			onClick={() => setClickedButton(applePay)}
			className="apple-pay-button wc-square-wallet-buttons"
		>
			<span className="text"></span>
			<span className="logo"></span>
		</div>
	);
```